src/scale/financetime.js

```diff
@@ -34,7 +34,7 @@ module.exports = function(d3_scale_linear, d3_time, d3_bisect, techan_util_rebin
      * @returns {*}
      */
     function scale(x, offset) {
-      var mappedIndex = dateIndexMap[+x];
+      var mappedIndex = dateIndexMap[x.getTime ? x.getTime() : +x];
```

Comment:
Nice catch here, but after profiling I would prefer:

```javascript
x instanceof Date ? x.getTime() : +x
```

Faster still (for me, linux chrome) and handles `undefined` and `null`s gracefully (without error). I'll add some test cases around the nulls and undefines.